modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -241,6 +241,14 @@ const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>): void => {
   const rng = editor.selection.getRng();
   const newBlockName = Options.getForcedRootBlock(editor);
 
+  const isInRoot = rng.collapsed && rng.startContainer === editor.dom.getRoot();
+  const start = SugarElement.fromDom(rng.startContainer);
+
+  const child = Traverse.child(start, rng.startOffset);
+  const isCefOpt = child.map((element) => SugarNode.isHTMLElement(element) && !ContentEditable.isEditable(element));
```

Comment:
Use exists or forall those use a predicate and return a boolean true or false by default.